src/Alert/index.jsx

```diff
@@ -1,37 +1,106 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
+import classNames from 'classnames';
 import Alert from 'react-bootstrap/Alert';
+import { useMediaQuery } from 'react-responsive';
+import useToggle from '../hooks/useToggle';
 import { Icon } from '..';
+import AlertButtons from './AlertButtons';
+import { alertButtonProps } from './propTypes';
+import { breakpoints } from '../Responsive';
 
 const WrappedAlert = React.forwardRef(({
   children,
   icon,
+  button,
+  dismissible,
+  onClose,
+  closeLabel,
+  show,
+  stacked,
   ...props
-}, ref) => (
-  <Alert
-    {...props}
-    className={`alert-content ${props.className}`}
-    ref={ref}
-  >
-    {icon && <Icon src={icon} className="alert-icon" />}
-    <div className="alert-message-content">
-      {children}
-    </div>
-  </Alert>
-));
+}, ref) => {
+  const [isOpen, /* open */, close] = useToggle(show);
```

Comment:
Is this component controlled, uncontrolled or either?

Reply:
Good catch. I've sort of made this both semi-controlled/uncontrolled simultaneously. I will update to make it controlled where the consumer must update the `show` prop when `onClose` is called on dismiss, like the react-bootstrap `Alert` does today.